virtual_ecosystem/models/soil/pools.py

```diff
@@ -755,3 +876,111 @@ def calculate_soil_nutrient_mineralisation(
 
     carbon_nutrient_ratio = pool_carbon / pool_nutrient
     return breakdown_rate / carbon_nutrient_ratio
+
+
+def calculate_nutrient_flows_to_necromass(
+    microbial_changes: MicrobialChanges, constants: SoilConsts
+) -> NDArray[np.float32]:
+    """Calculate the rate at which nutrients flow into the necromass pool.
+
+    These flows comprise of the nitrogen and phosphorus content of the dead cells and
+    denatured enzymes that flow into the necromass pool.
+
+    TODO - A core assumption here is that the stochiometry of the enzymes are identical
+    to the microbial cells. This assumption works for now but will have to be revisited
+    when fungi are added (as they have different stochiometric ratios but will
+    contribute to the same enzyme pools)
+
+    Args:
+        microbial_changes: Full set of changes to the microbial population due to
+            growth, death enzyme production, etc
+        constants: Set of constants for the soil model.
+
+    Returns:
+        The rate at which nitrogen is added to the necromass pool [kg N m^-3 day^-1]
+    """
+
+    return np.divide(
+        microbial_changes.necromass_generation, constants.microbial_c_n_ratio
+    )
```

Comment:
Out of curiosity, why not just using `/` to perform the division? Aren't these normal arrays already?

Reply:
Not actually sure, I just noticed that I was using `np.divide()` elsewhere and assumed that it was generically best practice, so used it here. Is the advantage of using `np.divide()` that it allows you to convert lists of numbers to `numpy` arrays?

Reply:
Pretty much. You can use array-like objects like lists with `np.divide` while you can only use proper arrays with `/`. I don't think it is a better practice, but rather serves to different purposes.